src/dash-table/dash/DataTable.js

```diff
@@ -1,44 +1,47 @@
-import * as R from 'ramda';
-import React, { Component } from 'react';
+import React, { Component, lazy, Suspense } from 'react';
 import PropTypes from 'prop-types';
+import LazyLoader from 'dash-table/LazyLoader';
+import LazyFactory from './LazyFactory';
+
+const isLazyComponentReady = (target, promise) => {
+    let resolve;
+    const isReady = new Promise(r => {
+        resolve = r;
+    });
+
+    const state = {
+        isReady,
+        get: lazy(() => {
+            return Promise.resolve(promise()).then(res => {
+                setTimeout(async () => {
+                    await resolve(true);
+                    state.isReady = true;
+                }, 0);
+
+                return res;
+            });
+        })
+    };
 
-import RealTable from 'dash-table/components/Table';
-
-import Logger from 'core/Logger';
-
-import genRandomId from 'dash-table/utils/generate';
-import isValidProps from './validate';
-import Sanitizer from './Sanitizer';
-
-/**
- * Dash DataTable is an interactive table component designed for
- * designed for viewing, editing, and exploring large datasets.
- * DataTable is rendered with standard, semantic HTML <table/> markup,
- * which makes it accessible, responsive, and easy to style. This
- * component was written from scratch in React.js specifically for the
- * Dash community. Its API was designed to be ergonomic and its behavior
- * is completely customizable through its properties.
- */
-export default class DataTable extends Component {
-    constructor(props) {
-        super(props);
-        let id;
-        this.getId = () => (id = id || genRandomId('table-'));
-        this.sanitizer = new Sanitizer();
-    }
+    Object.defineProperty(target, '_dashprivate_isLazyComponentReady', {
+        get: () => state.isReady
+    });
```

Comment:
This getter is what the renderer checks for before initiating callbacks https://github.com/plotly/dash/pull/899/files#diff-15692ba768eeab56e95aa409ab7c19b7R262